### PR TITLE
fix: add VRay system environment variable detection

### PR DESCRIPTION
**Pull request by leongdl**
### What was the problem/requirement? (What/Why)
- https://github.com/aws-deadline/deadline-cloud-for-3ds-max/issues
- When a user runs 3dsmax VRay without the environment variables set, the adapter does not give a clear indication why there is a failure.

### What was the solution? (How)
- Add an environment variable check before starting the adapter. Raise a runtime error, kill the adaptor w/ an appropriate error message.
- Handle different versions of 3dsmax by doing some release year parsing.

### What is the impact of this change?
- Better error handling and failure detection if the environment setup is not working well.

### How was this change tested?
- hatch build
- hatch run fmt
- hatch run lint
- hatch run test (New tests added)
- Running integration tests, with and without the environment variable set. All passed accordingly.

The error logs correctly show the missing environment variables are missing.
```
STDOUT: Performing action: {"name": "renderer", "args": {"renderer": "V_Ray_7_Hotfix_2"}}
STDOUT: Logger interceptor set up for render element manager
STDOUT: V-Ray renderer detected, but required environment variables are missing.
STDERR: Failed to initialize renderer: V-Ray renderer detected, but required environment variables are missing.
STDOUT: Please set the following variables in your host configuration script:
STDERR: Please set the following variables in your host configuration script:
STDOUT:   - VRAY_FOR_3DSMAX2026_MAIN
STDERR:   - VRAY_FOR_3DSMAX2026_MAIN
STDOUT: Logger interceptor torn down
STDERR: - 00:58:25.094 INFO: Begin resolve assembly...
STDOUT: Performing action: {"name": "close", "args": null}
```

### Was this change documented?
- This is an error message.

### Is this a breaking change?
- No.

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

### CODE REVIEW

File: src/deadline/max_adaptor/MaxClient/max_client.py [hunk 1]
```diff
@@ -81,9 +81,14 @@ def set_renderer(self, renderer: dict):
...
         self.logger_interceptor.setup()
 
-        render_handler = get_render_handler(renderer["renderer"])
-
-        self.actions.update(render_handler.action_dict)
+        try:
+            render_handler = get_render_handler(renderer["renderer"])
+            self.actions.update(render_handler.action_dict)
+        except Exception as e:
+            # Log the error and close Max to ensure clean exit
```
Comment (leongdl) [inline src/deadline/max_adaptor/MaxClient/max_client.py:88]: It was not gracefully shutting down before, so this makes sure the system exits, and can die from the server side.

File: src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py [hunk 2]
```diff
@@ -24,6 +26,44 @@ def __init__(self, gpu):
...
+            f"VRAY_MDL_PATH_3DSMAX{year}",
+        ]
+
+        # Check for missing variables
+        missing_vars = [var for var in required_vars if var not in os.environ]
+
+        if missing_vars:
+            error_msg = (
+                f"V-Ray renderer detected, but required environment variables are missing.\n"
+                f"Please set the following variables in your host configuration script:\n"
```
Comment (jairaws) [inline src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py]: "host configuration script" -> Seems very specific. Not all customers will use 3dsMax with a host configuration script.

File: src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py [hunk 3]
```diff
@@ -24,6 +26,44 @@ def __init__(self, gpu):
...
         self.gpu: bool = gpu
+        self._validate_vray_environment()
+
+    def _validate_vray_environment(self) -> None:
+        """
+        Validates that required VRay environment variables are set.
+        Raises RuntimeError with actionable message if variables are missing.
+        """
+        # Get 3ds Max year using the same formula as max_render_submitter.py
+        max_version = rt.maxVersion()
```
Comment (jairaws) [inline src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py]: nit: type

Same for the other variables.

File: src/deadline/max_adaptor/MaxClient/max_client.py [hunk 4]
```diff
@@ -81,9 +81,14 @@ def set_renderer(self, renderer: dict):
...
+            logger.error(f"Failed to initialize renderer: {e}")
+            self.close()
```
Comment (jericht) [inline src/deadline/max_adaptor/MaxClient/max_client.py:90]: Was the 3dsMax process being orphaned without this?

File: src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py [hunk 5]
```diff
@@ -24,6 +26,44 @@ def __init__(self, gpu):
...
+        ]
+
+        # Check for missing variables
+        missing_vars = [var for var in required_vars if var not in os.environ]
+
+        if missing_vars:
+            error_msg = (
+                f"V-Ray renderer detected, but required environment variables are missing.\n"
+                f"Please set the following variables in your host configuration script:\n"
+                f"{chr(10).join(f'  - {var}' for var in missing_vars)}"
```
Comment (jericht) [inline src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py]:
```suggestion
                f"{os.linesep.join(f'  - {var}' for var in missing_vars)}"
```

nit: this is a bit easier to read, same for below message

File: test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+        """Test VRay GPU handler also validates environment"""
+        from deadline.max_adaptor.MaxClient.render_handlers.vray_handler import VrayHandler
+
+        # Mock 3ds Max 2026: #(28000, 28, 0, 0, 0)
+        mock_rt.maxVersion.return_value = [28000, 28, 0, 0, 0]
+
+        # Should not raise
+        handler = VrayHandler(gpu=True)
+        assert handler is not None
+        assert handler.gpu is True
```
Comment (jericht) [inline test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py]: Could this test case just be a parameterization on `gpu` for the above test case? `test_vray_handler_with_all_env_vars`

File: test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+                # Mock 3ds Max 2025: #(27000, 27, 0, 0, 0)
+                mock_rt.maxVersion.return_value = [27000, 27, 0, 0, 0]
+
+                # Clear the module cache to ensure fresh import after patching
+                if "deadline.max_adaptor.MaxClient.render_handlers.vray_handler" in sys.modules:
+                    del sys.modules["deadline.max_adaptor.MaxClient.render_handlers.vray_handler"]
+
+                from deadline.max_adaptor.MaxClient.render_handlers.vray_handler import (
+                    VrayHandler,
+                )
```
Comment (jericht) [inline test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py]: Do we need to do this for the test? If it's because the empty `os.environ` isn't taking effect, we can try patching it at the call site:

```py
with patch.dict("deadline.max_adaptor.MaxClient.render_handlers.vray_handler.os.environ", {}, clear=True)
```

File: test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+
+                # Expect RuntimeError
+                with pytest.raises(RuntimeError) as exc_info:
+                    VrayHandler(gpu=False)
+
+                # Verify error message contains the missing variable
+                error_msg = str(exc_info.value)
+                assert "VRAY_MDL_PATH_3DSMAX2025" in error_msg
```
Comment (jericht) [inline test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py]: nit: we can probably combine these tests into one and parametrize on which env var we are missing

File: test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+                # Expect RuntimeError
+                with pytest.raises(RuntimeError) as exc_info:
+                    VrayHandler(gpu=False)
+
+                # Verify error message contains only the missing variable
+                error_msg = str(exc_info.value)
+                assert "VRAY_MDL_PATH_3DSMAX2025" in error_msg
+                # Present variables should not be in the error
+                assert "VRAY_FOR_3DSMAX2025_MAIN" not in error_msg
+                assert "VRAY_FOR_3DSMAX2025_PLUGINS" not in error_msg
```
Comment (jericht) [inline test/unit/deadline_adaptor_for_3ds_max/MaxClient/render_handlers/test_vray_handler.py]: This can also be combined into a parametrized test for missing env vars

File: src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py (hunk 3 above)
Comment (leongdl) [reply src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py]: sure, easy changes

File: src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py (hunk 2 above)
Comment (leongdl) [reply src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py]: Sure, I'll change it to "system environment variable"

File: src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py (hunk 5 above)
Comment (leongdl) [reply src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py]: sounds good.

File: src/deadline/max_adaptor/MaxClient/max_client.py (hunk 4 above)
Comment (leongdl) [reply src/deadline/max_adaptor/MaxClient/max_client.py:90]: Yup - because this is on the client. The server keeps sending commands hoping it continues.

File: src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py (hunk 3 above)
Comment (leongdl) [reply src/deadline/max_adaptor/MaxClient/render_handlers/vray_handler.py]: I need to add a .kiro/steering!!

I keep forgetting. Will add it as a separate PR.

**Review by jairaws — APPROVED** [review]

**Review by jericht — APPROVED** [review]
